lib/directory-tree.js

```diff
@@ -48,13 +57,23 @@ function directoryTree (path, options, onEachFile) {
 		}
 	}
 	else if (stats.isDirectory()) {
-		let dirData = safeReadDirSync(path);
-		if (dirData === null) return null;
-		
-		item.children = dirData
-			.map(child => directoryTree(PATH.join(path, child), options, onEachFile))
-			.filter(e => !!e);
-		item.size = item.children.reduce((prev, cur) => prev + cur.size, 0);
+        if(depth<options.depth || options.depth == -1){
```

Comment:
This is a recursive function. Let's extract the base case at the beginning of it. I.e. at the beginning of the function, if you don't have any more "depth" to explore, stop.